src/System.Net.Sockets.Legacy/tests/FunctionalTests/DnsEndPointTest.cs

```diff
@@ -58,8 +60,8 @@ public void Socket_ReceiveFromDnsEndPoint_ArgumentException()
         {
             using (Socket sock = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp))
             {
-                sock.Bind(new IPEndPoint(IPAddress.Loopback, 8080));
-                EndPoint endpoint = new DnsEndPoint("localhost", 8080);
+                sock.Bind(new IPEndPoint(IPAddress.Loopback, TestPortBase + 4));
```

Comment:
Previously, all tests were using the same port. The problem appeared in the initial ToF attempt where tests were executed in parallel. 

The increment (interlocked or not) is still as fragile as this one as any process can take over the same port.

There is another method where you specify port 0 and the OS automatically binds to the next available ephemeral port. The problem is that the code will need to make a syscall to find the actual port (see the Security tests). This isn't really a common scenario except for FTP but solves this particular problem.

I think that most tests would work fine with this dynamic port allocation system although I would still leave a few on fixed ports.